pkg/controller/rebuild.go

```diff
@@ -53,12 +53,12 @@ func (c *Controller) VerifyRebuildReplica(address string) error {
 		return fmt.Errorf("invalid mode %v for replica %v to check", replica.Mode, address)
 	}
 
-	fromDisks, _, err := GetReplicaDisksAndHead(rwReplica.Address)
+	fromDisks, _, err := GetReplicaDisksAndHead(rwReplica.Address, c.Name, "")
 	if err != nil {
 		return err
 	}
 
-	toDisks, _, err := GetReplicaDisksAndHead(address)
+	toDisks, _, err := GetReplicaDisksAndHead(address, c.Name, instanceName)
```

Comment:
You mentioned it in the instance-manager PR: https://github.com/longhorn/longhorn-instance-manager/pull/229#discussion_r1251101973.

This is something @PhanLe1010 and I have discussed quite a bit, including in https://github.com/longhorn/longhorn-engine/pull/902#discussion_r1267028573.

Adding instanceName to types.Replica would be one thing, but we would need to make quite a few changes to controller client methods (which would propagate up the Longhorn stack) in order to make it useful. Because of the added complexity, and the limited practical benefit, we have steered away from doing this. If we do proceed that direction, I suggest we do it in a separate set of PRs.

Changing to VolumeName in the latest commit.

Reply:
> Because of the added complexity, and the limited practical benefit, we have steered away from doing this. If we do proceed that direction, I suggest we do it in a separate set of PRs.

Sometimes this lacking of the replica name makes the log debugging of the engine more difficult. And now with this info, we can make the identity validation between engines and replicas more convenient. To be honest, I prefer to have this change. WDYT? cc @innobead

Reply:
YES. After adding it, we should input `<replica name>=<replica address>` for the CLI and a map for gRPC API `VolumeStart`. It's more trivial and complicated.

Reply:
Agree that we should handle it in a new improvement ticket since this PR already satisfy its goal